damnit/backend/extract_data.py

```diff
@@ -198,51 +170,139 @@ def update_db_vars(self):
             ))
         self.kafka_prd.flush()
 
-    def extract_and_ingest(self, proposal, run, cluster=False,
-                           run_data=RunData.ALL, match=(), variables=(), mock=False):
-        if proposal is None:
-            proposal = self.db.metameta['proposal']
+class RunExtractor(Extractor):
+    def __init__(self, proposal, run, cluster=False, run_data=RunData.ALL,
+                 match=(), variables=(), mock=False):
+        super().__init__()
+        self.proposal = proposal
+        self.run = run
+        self.cluster = cluster
+        self.run_data = run_data
+        self.match = match
+        self.variables = variables
+        self.mock = mock
+        self.uuid = str(uuid4())
+        self.running_msg = msg_dict(MsgKind.processing_running, {
+            'processing_id': self.uuid,
+            'proposal': proposal,
+            'run': run,
+            'data': run_data.value,
+            'hostname': socket.gethostname(),
+            'username': getuser(),
+            'slurm_cluster': self._slurm_cluster(),
+            'slurm_job_id': self._slurm_job_id(),
+        })
+
+    @staticmethod
+    def _slurm_cluster():
+        # For some reason, SLURM_CLUSTER_NAME is '(null)'. This is a workaround:
+        if not os.environ.get('SLURM_JOB_ID', ''):
+            return None
+        partition = os.environ.get('SLURM_JOB_PARTITION', '')
+        return 'solaris' if (partition == 'solcpu') else 'maxwell'
+
+    @staticmethod
+    def _slurm_job_id():
+        # Get Slurm job ID in the same format that squeue uses
+        array_job_id = os.environ.get('SLURM_ARRAY_JOB_ID', '')
+        array_task_id = os.environ.get('SLURM_ARRAY_TASK_ID', '')
+        if array_job_id and array_task_id:
+            # In an array job, e.g. '12380337_308'
+            return f"{array_job_id}_{array_task_id}"
+        else:
+            # Not an array job - just use the regular job ID
+            slurm_job_id = os.environ.get('SLURM_JOB_ID', '')
```

Comment:
```suggestion
            return slurm_job_id = os.environ.get('SLURM_JOB_ID', '')
```

Is this branch used at all or all jobs are started as array now?